Kernel/Net/NetworkAdapter.cpp

```diff
@@ -129,6 +129,44 @@ void NetworkAdapter::send_ipv4(const MACAddress& destination_mac, const IPv4Addr
     send_raw((const u8*)&eth, ethernet_frame_size);
 }
 
+void NetworkAdapter::send_ipv4_fragmented(const MACAddress& destination_mac, const IPv4Address& destination_ipv4, IPv4Protocol protocol, const u8* payload, size_t payload_size, u8 ttl)
+{
+    // packets must be split on the 64-bit boundary
+    int packet_boundary_size = (mtu() - sizeof(IPv4Packet) - sizeof(EthernetFrameHeader)) & 0xfffffff8;
+    int fragment_block_count = (payload_size + packet_boundary_size) / packet_boundary_size;
+    int last_block_size = payload_size - packet_boundary_size * (fragment_block_count - 1);
+    int number_of_blocks_in_fragment = packet_boundary_size / 8;
+
+    auto identification = m_bytes_out; // FIXME: somehow generate a true random ID
```

Comment:
Can we just call `get_good_random()` here instead? It's available in `<Kernel/Random.h>`

Reply:
Honestly, I am not entirely sure about this identifier field.
The text of the RFC says that it has to be _unique_ per src/dst and protocol; however in practice, it seems to me that everyone just assigns randoms.

In the reassembly process, I don't see an issue with it being random, but that branch is super duper WIP right now, so I can't tell for sure.